packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart

```diff
@@ -78,6 +79,22 @@ If this option is not provided, the version number will be determined from the p
         'staging',
         negatable: false,
         help: 'Whether to publish the patch to the staging environment.',
+      )
+      ..addFlag(
+        'sign',
+        help: r'''
+When true will sign allow the hash of the patch to be signed.
+
+The CLI will look for a script called sign_hash in the current directory, which
+is expected to receive the generated hash, and print the signed hash to stdout.
+
+A naive implementation of sign_hash in BASH might look like this:
+
+```bash
+echo $(echo "$1" | openssl dgst -sha256 -sign ./private.pem /dev/stdin | base64)
```

Comment:
The idea behind this approach is that we have flexibility for now in the phases where we are testing the new feature.

By hooking our CLI in a generic script that its protocol is simply to receive the hash and print the signature in the stdout we test generating the signature from the `openssl` command, and we can also test hooking up a signature service, where we would just replace the script to be one that integrates with the signature service that we want to test, etc.

Basically I tried to think in a way that will unblock us to test different approaches for the patch signing, without needing to keep us changing the CLI.

I am not attached to this approach though, so if you feel there are any issues, or if it seems too orthodox, let me know and I am happy to change it.